후/src/mission2_0/RandomGacha.java

```diff
@@ -0,0 +1,82 @@
+package mission2_0;
+
+import java.util.*;
+import java.util.stream.Stream;
+
+class RandomGacha {
+
+    final static int MIN_COIN = 1;
+
+    Scanner coinSlot;
+    private int coin = 0;
+    List<CocoaMember> members;
+
+    RandomGacha() {
+        coinSlot = new Scanner(System.in);
+        this.members = CocoaMember.makeShuffledSquad();
+    }
+
+    boolean isValidInput(String input) {
+        try {
+            if (Integer.parseInt(input) < MIN_COIN) {
+                throw new InputMismatchException();
+            }
+            return true;
+        } catch (NumberFormatException e) {
+            System.out.println("숫자로 입력");
+            return false;
+        } catch (InputMismatchException e) {
+            System.out.println("0보다 큰 숫자로 입력");
+            return false;
+        }
+    }
+
+    void getUserCoin() {
+        String input = "";
+
+        System.out.print("insert coin >>>>> ");
+        input = coinSlot.nextLine();
+
+        if (isValidInput(input)) {
+            int coins = Integer.parseInt(input);
+            insertCoin(coins);
+            coinSlot.close();
+            return;
+        }
+
+        getUserCoin();
+    }
+
+    void insertCoin(int coins) {
+        this.coin += coins;
+    }
+
+    void pick() {
+        StringBuilder result = new StringBuilder();
+        Stream<CocoaMember> memberStream;
+        memberStream = members.stream();
+
+        memberStream
+            .limit(coin)
+            .peek(member -> coin--)
+            .forEach(member -> result.append(member.getName() + ", "));
```

Comment:
```suggestion
        List<CocoaMember> subMembers = members.subList(0, Math.min(coin, members.size()));
        for (CocoaMember member: subMembers) {
            result.append(member.getName() + ", ");
            coin--;
        }
        System.out.println(result.substring(0, result.length() - 2));
```

멤버 스트림, 코인 스트림 이렇게 스트림 두개 생성하는 것보다 List하나 생성해서 작업하는 것이 나을 것 같아서 이렇게 고쳐보았습니다..

Reply:
https://github.com/who-hoo/cocoa2021-java-week1-squad1/pull/14#discussion_r742917319

참고해보시면 될 것 같아요.